### PR TITLE
Moving new-kernel to 5.10.1 LTS

### PR DESCRIPTION
**Pull request by rvs**
Signed-off-by: Roman Shaposhnik <rvs@zededa.com>

### CODE REVIEW

File: pkg/new-kernel/rtl8821CU.patch [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+ 
+-			oldfs = get_fs();
+ 			ret = writeFile(fp, buf, sz);
+-			set_fs(oldfs);
+ 			closeFile(fp);
+ 
+ 			RTW_INFO("%s writeFile, ret:%d\n", __FUNCTION__, ret);
+-- 
+2.20.1
+
```
Comment (temerkhanov) [inline pkg/new-kernel/rtl8821CU.patch:76]: This rtl8821CU needs more attention, most probably. Its readFile()/writeFile functions probably access data from the userspace, that's there get_kernel_nofault()/get_user() are supposed to be used

Comment (rvs) [reply pkg/new-kernel/rtl8821CU.patch:76]: Yeah -- I totally agree @temerkhanov -- this is just a quick hack to get it built. Can you help making sure we don't break it in the process?

**Comment by Insei** [comment]
kernel branch 5.10 - LTS? I see that 5.10 is STABLE, not LTS.
https://www.kernel.org/

EVE use LTS branches for kernel or STABLE in releases?

**Comment by rvs** [comment]
> kernel branch 5.10 - LTS? I see that 5.10 is STABLE, not LTS.
> https://www.kernel.org/


5.10 is going to be an LTS -- they just haven't gotten to properly labeling it yet. https://www.osnews.com/story/132730/linux-kernel-5-10-lts-released/

> EVE use LTS branches for kernel or STABLE in releases?

LTS

File: pkg/new-kernel/kernel_config-5.10.x-aarch64 [hunk 2]
```diff
@@ -5947,8 +6215,27 @@ CONFIG_CRYPTO_JITTERENTROPY=y
...
+# CONFIG_CRYPTO_LIB_CHACHA is not set
+# CONFIG_CRYPTO_LIB_CURVE25519 is not set
+CONFIG_CRYPTO_LIB_DES=y
+CONFIG_CRYPTO_LIB_POLY1305_RSIZE=9
+# CONFIG_CRYPTO_LIB_POLY1305 is not set
+# CONFIG_CRYPTO_LIB_CHACHA20POLY1305 is not set
+CONFIG_CRYPTO_LIB_SHA256=y
 CONFIG_CRYPTO_HW=y
-# CONFIG_CRYPTO_DEV_MARVELL_CESA is not set
+CONFIG_CRYPTO_DEV_ALLWINNER=y
```
Comment (temerkhanov) [inline pkg/new-kernel/kernel_config-5.10.x-aarch64:6235]: Do we have it?

File: pkg/new-kernel/kernel_config-5.10.x-aarch64 [hunk 3]
```diff
@@ -5959,16 +6246,23 @@ CONFIG_CRYPTO_HW=y
...
+# CONFIG_CRYPTO_DEV_ZYNQMP_AES is not set
 CONFIG_CRYPTO_DEV_VIRTIO=y
 CONFIG_CRYPTO_DEV_BCM_SPU=m
 # CONFIG_CRYPTO_DEV_SAFEXCEL is not set
 # CONFIG_CRYPTO_DEV_CCREE is not set
 # CONFIG_CRYPTO_DEV_HISI_SEC is not set
+# CONFIG_CRYPTO_DEV_HISI_SEC2 is not set
 # CONFIG_CRYPTO_DEV_HISI_ZIP is not set
+# CONFIG_CRYPTO_DEV_HISI_HPRE is not set
+CONFIG_CRYPTO_DEV_AMLOGIC_GXL=m
```
Comment (temerkhanov) [inline pkg/new-kernel/kernel_config-5.10.x-aarch64:6264]: Do we have it anywhere?

File: pkg/new-kernel/kernel_config-5.10.x-aarch64 [hunk 4]
```diff
@@ -5381,14 +5630,17 @@ CONFIG_RESET_TEGRA_BPMP=y
...
 # CONFIG_PHY_SUN6I_MIPI_DPHY is not set
 # CONFIG_PHY_SUN9I_USB is not set
+# CONFIG_PHY_SUN50I_USB3 is not set
 # CONFIG_PHY_MESON8B_USB2 is not set
 # CONFIG_PHY_MESON_GXL_USB2 is not set
-# CONFIG_PHY_MESON_GXL_USB3 is not set
 # CONFIG_PHY_MESON_G12A_USB2 is not set
 # CONFIG_PHY_MESON_G12A_USB3_PCIE is not set
+CONFIG_PHY_MESON_AXG_PCIE=y
+CONFIG_PHY_MESON_AXG_MIPI_PCIE_ANALOG=y
```
Comment (temerkhanov) [inline pkg/new-kernel/kernel_config-5.10.x-aarch64:5643]: Do we have these?

File: pkg/new-kernel/kernel_config-5.10.x-aarch64 [hunk 5]
```diff
@@ -5350,24 +5593,30 @@ CONFIG_MVEBU_ICU=y
...
 CONFIG_MVEBU_PIC=y
 CONFIG_MVEBU_SEI=y
+CONFIG_LS_EXTIRQ=y
 CONFIG_LS_SCFG_MSI=y
 CONFIG_PARTITION_PERCPU=y
 CONFIG_QCOM_IRQ_COMBINER=y
 CONFIG_MESON_IRQ_GPIO=y
 CONFIG_QCOM_PDC=y
 CONFIG_IMX_IRQSTEER=y
+CONFIG_IMX_INTMUX=y
```
Comment (temerkhanov) [inline pkg/new-kernel/kernel_config-5.10.x-aarch64:5603]: These two are for the i.MX family

File: pkg/new-kernel/kernel_config-5.10.x-aarch64 [hunk 6]
```diff
@@ -5189,15 +5419,17 @@ CONFIG_FSL_DPAA=y
...
 # CONFIG_DPAA2_CONSOLE is not set
+# CONFIG_FSL_RCPM is not set
 # end of NXP/Freescale QorIQ SoC drivers
 
 #
 # i.MX SoC drivers
 #
 CONFIG_IMX_GPCV2_PM_DOMAINS=y
-# CONFIG_IMX_SCU_SOC is not set
+CONFIG_SOC_IMX8M=y
```
Comment (temerkhanov) [inline pkg/new-kernel/kernel_config-5.10.x-aarch64:5432]: Do we support i.MX8?

File: pkg/new-kernel/kernel_config-5.10.x-aarch64 [hunk 7]
```diff
@@ -3329,14 +3436,18 @@ CONFIG_PINCTRL_TEGRA_XUSB=y
 #
 CONFIG_EINT_MTK=y
```
Comment (temerkhanov) [inline pkg/new-kernel/kernel_config-5.10.x-aarch64:3437]: The whole section is for Mediatek. Do we support it?

File: pkg/new-kernel/kernel_config-5.10.x-aarch64 [hunk 8]
```diff
@@ -1705,9 +1775,26 @@ CONFIG_PCIE_ARMADA_8K=y
...
 CONFIG_PCIE_HISI_STB=y
 CONFIG_PCI_MESON=y
-# CONFIG_PCIE_TEGRA194 is not set
+# CONFIG_PCIE_TEGRA194_HOST is not set
+# CONFIG_PCIE_TEGRA194_EP is not set
 CONFIG_PCIE_AL=y
 # end of DesignWare PCI Core Support
+
+#
+# Mobiveil PCIe Core Support
```
Comment (temerkhanov) [inline pkg/new-kernel/kernel_config-5.10.x-aarch64:1784]: Can we find it anywhere?

File: pkg/new-kernel/rtl8821CU.patch (hunk 1 above)
Comment (temerkhanov) [reply pkg/new-kernel/rtl8821CU.patch:76]: This should work though - reading files through f_ops should do all the necessary steps

**Comment by Insei** [comment]
> > kernel branch 5.10 - LTS? I see that 5.10 is STABLE, not LTS.
> > https://www.kernel.org/
> 
> 5.10 is going to be an LTS -- they just haven't gotten to properly labeling it yet. https://www.osnews.com/story/132730/linux-kernel-5-10-lts-released/
> 
> > EVE use LTS branches for kernel or STABLE in releases?
> 
> LTS

My bad, thx

File: pkg/new-kernel/rtl8821CU.patch (hunk 1 above)
Comment (rvs) [reply pkg/new-kernel/rtl8821CU.patch:76]: thanks for confirming @temerkhanov !

**Review by rvs — COMMENTED** [review]
Hey @temerkhanov -- I'm doing a few additional tweaks for the ARM64 config (based on your feedback) and then will be pushing it.

**Comment by temerkhanov** [comment]
We will also need to re-enable io_uring in the master branch

**Comment by rvs** [comment]
True @temerkhanov -- can you please send a pull request for that?

**Comment by temerkhanov** [comment]
https://github.com/lf-edge/eve/pull/1761

**Pull request merged**